### PR TITLE
Wip/hughsie/invalid compressed data

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: contrib/fwupd.spec.in [hunk 1]
```diff
@@ -269,6 +269,11 @@ mkdir -p --mode=0700 $RPM_BUILD_ROOT%{_localstatedir}/lib/fwupd/gnupg
 %post
 %systemd_post fwupd.service
 
+# change vendor-installed remotes to use the default keyring type
+for fn in /etc/fwupd/remotes.d/*.conf;
+    do sed -i 's/Keyring=gpg/#Keyring=pkcs/g' "$fn";
+done
```
Comment (superm1) [inline contrib/fwupd.spec.in:275]: I wonder if we also want this in other packaging types too?  Or possibly it should be code in the daemon to do it?

**Review by superm1 — APPROVED** [review]
The two fix commits are fine, but question on the packaging change.

File: contrib/fwupd.spec.in (hunk 1 above)
Comment (hughsie) [reply contrib/fwupd.spec.in:275]: It can't be in the Daemon as in the snap case the Daemon is the old one, not the new one.

**Comment by hughsie** [comment]
I think the premise of this PR is wrong. I think if was want/have to support new-lib, old-daemon then we have to go back to constructing a fake jcat entry if passed a asc/pkcs file in the engine.

**Comment by hughsie** [comment]
I'll merge as it fixes a lot of the immediate problems people are seeing but I do think we need the fake jcat thing for completeness. And maybe snap/deb versions of that sed line too.

**Pull request merged**